### PR TITLE
Show explicit error message when extracting interface from module

### LINKED ISSUE #6754: Extract Interface in VB module gives misleading error message
The existing error message makes me thing I merely need to make my methods extractable (instead of telling me not to extract interface on a module)

image

### PR DESCRIPTION
**Pull request by ryzngard**
Fixes #6754

image

### CODE REVIEW

File: src/Features/VisualBasic/Portable/ExtractInterface/VisualBasicExtractInterfaceService.vb [hunk 1]
```diff
@@ -158,5 +159,15 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.ExtractInterface
 
             Return CreateFinalSolution(unformattedSolution, documentIds, docToRootMap)
         End Function
+
+        Friend Overrides Function TryGetLanguageSpecificErrorMessage(containingType As ISymbol, <NotNullWhen(True)> ByRef errorMessage As String) As Boolean
+
+            If TryCast(containingType, IModuleSymbol) Is Nothing Then
+                errorMessage = VBFeaturesResources.Could_not_extract_interface_colon_Cannot_extract_interface_for_modules
```
Comment (CyrusNajmabadi) [inline src/Features/VisualBasic/Portable/ExtractInterface/VisualBasicExtractInterfaceService.vb:166]: slightly confused.  teh test says "if the containing is *not* a module".  shouldn't this be `is not nothing`? or `typeof containingtype is imodulesymbol`?

File: src/Features/VisualBasic/Portable/VBFeaturesResources.resx [hunk 2]
```diff
@@ -1224,4 +1224,8 @@ Sub(&lt;parameterList&gt;) &lt;statement&gt;</value>
   <data name="Project_level_Imports" xml:space="preserve">
     <value>Project level 'Imports'</value>
   </data>
+  <data name="Could_not_extract_interface_colon_Cannot_extract_interface_for_modules" xml:space="preserve">
+    <value>Could not extract interface: Cannot extract interface for modules</value>
```
Comment (CyrusNajmabadi) [inline src/Features/VisualBasic/Portable/VBFeaturesResources.resx:1228]: phrase seems redundant.  can it just be `Can not extract interface for modules`?

Comment (ryzngard) [reply src/Features/VisualBasic/Portable/VBFeaturesResources.resx:1228]: It just follows the other phrasing we use. Happy to change to just "Can not extract interface for modules" though if that's preferred

Comment (CyrusNajmabadi) [reply src/Features/VisualBasic/Portable/VBFeaturesResources.resx:1228]: if this is consistent with the other strings, then i'm fine keeping as is :)

File: src/Features/VisualBasic/Portable/ExtractInterface/VisualBasicExtractInterfaceService.vb (hunk 1 above)
Comment (ryzngard) [reply src/Features/VisualBasic/Portable/ExtractInterface/VisualBasicExtractInterfaceService.vb:166]: hmm true, need to revisit why it passed as expected with manual testing. Maybe I was changing to be more succinct VB and forgot the `Not`

**Pull request closed without merging**